packages/go_router/lib/src/match.dart

```diff
@@ -34,7 +32,7 @@ class RouteMatch {
         subloc: restLoc,
         extra: extra,
         error: null,
-        pageKey: ValueKey<String>(route.hashCode.toString()),
+        pageKey: PageKey(path: route.hashCode.toString()),
```

Comment:
Why create a new subclass where does the same thing a value key? the path: parameter here doesn't make sense if we use hashcode here

Reply:
It was to be able to compare the path between 2 keys to know whether they were equal so we can reuse the same key or not.
But I removed it now